File: JavaAlgorithm/src/main/java/algo/tzashinorpu/SpecialSubject/TopInterview2Round/maxDepth_104.java
package algo.tzashinorpu.SpecialSubject.TopInterview2Round;

import algo.tzashinorpu.SpecialSubject.TreeNode;

import java.util.ArrayDeque;

public class maxDepth_104 {
    public int maxDepth(TreeNode root) {
        int level = 0;
        ArrayDeque<TreeNode> deque = new ArrayDeque<>();
        if (root != null) deque.add(root);
        while (!deque.isEmpty()) {
            level++;
            int size = deque.size();
            for (int i = 0; i < size; i++) {
                TreeNode poll = deque.pollFirst();
                if (poll.left != null) deque.offerLast(poll.left);
                if (poll.right != null) deque.offerLast(poll.right);
            }
        }
        return level;
    }
}

File: Clion/labuladong/chapter0/binary_tree/maxDepth_104.cpp
//
// Created by tseringnorpu on 6/3/2023.
//
#include "../../header/header.h"
using namespace labuladong;

int Solution::maxDepth(TreeNode *root) {
    if(!root) return 0;
    int l=maxDepth(root->left)+1;
    int r = maxDepth(root->right)+1;
    return max(l, r);
}